modules/base/include/modules/base/processors/inputselector.h

```diff
@@ -130,7 +130,12 @@ InputSelector<Inport, Outport>::InputSelector()
 
 template <typename Inport, typename Outport>
 void InputSelector<Inport, Outport>::process() {
-    outport_.setData(inport_.getVectorData().at(selectedPort_.get()));
+    auto data = inport_.getVectorData();
+    if (selectedPort_.get() < data.size()) {
+        outport_.setData(data[selectedPort_.get()]);
+    } else {
+        outport_.setData(data.back());
```

Comment:
any reason here for using the last one?

Reply:
I would be the "closest" one the the selected...